### PR TITLE
feat: Addition of fields for reCAPTCHA

### LINKED ISSUE #6043: Add fields for reCAPTCHA Keys
Sub-Issue Of #6036

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #6043 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Changes proposed in this pull request:

- Addition of fields for reCAPTCHA

### CODE REVIEW

**Comment by kushthedude** [comment]
@iamareebjamal @CosmicCoder96 @niranjan94 Please Review

File: app/api/schema/settings.py [hunk 1]
```diff
@@ -129,6 +129,13 @@ class Meta:
     gs_secret = fields.Str(allow_none=True)
     gs_bucket_name = fields.Str(allow_none=True)
 
+    #
+    # Google reCAPTCHA
+    #
+
+    captcha_site_key = fields.Str(allow_none=True)
```
Comment (iamareebjamal) [inline app/api/schema/settings.py]: Add a boolean field - google_captcha_enabled

Comment (kushthedude) [reply app/api/schema/settings.py]: Doing the following

Comment (kushthedude) [reply app/api/schema/settings.py]: @iamareebjamal Please Review now

**Comment by kushthedude** [comment]
@iamareebjamal @CosmicCoder96 @niranjan94 Please have a look

File: migrations/versions/7388e7504caa_.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# revision identifiers, used by Alembic.
+revision = '7388e7504caa'
+down_revision = '09c86e38f5e7'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('settings', sa.Column('google_recaptcha_secret_key', sa.String(), nullable=True))
+    op.add_column('settings', sa.Column('google_recaptcha_site_key', sa.String(), nullable=True))
+    op.add_column('settings', sa.Column('is_google_recaptcha_enabled', sa.Boolean(), nullable=False))
```
Comment (iamareebjamal) [inline migrations/versions/7388e7504caa_.py]: This will fail as there's no default set

**Comment by iamareebjamal** [comment]
Solve hound issues

**Comment by kushthedude** [comment]
@iamareebjamal Please Check Now

**Comment by kushthedude** [comment]
@iamareebjamal @CosmicCoder96 @niranjan94 Please review

File: app/api/schema/settings.py [hunk 3]
```diff
@@ -52,6 +52,11 @@ class Meta:
     cookie_policy = fields.Str(allow_none=True)
     cookie_policy_link = fields.Str(allow_none=True)
 
+    #
+    # CAPTCHA
+    #
+    is_google_recaptcha_enabled = fields.Bool(dump_only=True)
```
Comment (iamareebjamal) [inline app/api/schema/settings.py]: Why is it not with other properties?

File: app/models/setting.py [hunk 4]
```diff
@@ -47,6 +47,14 @@ class Setting(db.Model):
     gs_secret = db.Column(db.String)
     gs_bucket_name = db.Column(db.String)
 
+    #
+    # CAPTCHA
+    #
+
+    # Google reCAPTCHA
+    google_recaptcha_site = db.Column(db.String)
```
Comment (iamareebjamal) [inline app/models/setting.py]: Where's the boolean column?

**Comment by iamareebjamal** [comment]
Please first review the PRs yourself by seeing the changes

File: app/api/schema/settings.py (hunk 3 above)
Comment (kushthedude) [reply app/api/schema/settings.py]: @iamareebjamal I saw the way how various payment gateways were implemented , None of their boolean was defined in `Settings Model` , So thought to go for the same ?

Comment (iamareebjamal) [reply app/api/schema/settings.py]: We should not repeat our mistakes

Comment (iamareebjamal) [reply app/api/schema/settings.py]: The reason for adding an enabled field is that if I want to disable the thing temporarily, or let's say have to remove the captcha or a payment system, I should not need to remove the API keys for them

Comment (kushthedude) [reply app/api/schema/settings.py]: > We should not repeat our mistakes

That was cool, 

> The reason for adding an enabled field is that if I want to disable the thing temporarily, or let's say have to remove the captcha or a payment system, I should not need to remove the API keys for them

Got it, will make appropriate changes

**Comment by mrsaicharan1** [comment]
About the boolean columns, https://github.com/fossasia/open-event-frontend/blob/development/app/models/setting.js#L46,
@kushthedude They are usually computed properties which can be toggled on the frontend, if that helps. They are implemented in the previous gateways too, such as `is_stripe_activated` &`is_paypal_activated` which are toggle buttons

**Comment by kushthedude** [comment]
@iamareebjamal Did the requested changes, Please review.

**Comment by kushthedude** [comment]
@iamareebjamal Did the requested changes, Please review.

File: migrations/versions/232b13d4e86a_.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+# revision identifiers, used by Alembic.
+revision = '232b13d4e86a'
+down_revision = '14e31b3a860b'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('settings', sa.Column('google_recaptcha_secret', sa.String(), nullable=True))
+    op.add_column('settings', sa.Column('google_recaptcha_site', sa.String(), nullable=True))
+    op.add_column('settings', sa.Column('is_google_recaptcha_enabled', sa.Boolean(), nullable=False))
```
Comment (iamareebjamal) [inline migrations/versions/232b13d4e86a_.py]: This migration will fail

Comment (kushthedude) [reply migrations/versions/232b13d4e86a_.py]: @iamareebjamal i specified the default value above , its not showing in migration file , what should i do?

Comment (iamareebjamal) [reply migrations/versions/232b13d4e86a_.py]: See previous PRs where default has been supplied

**Comment by kushthedude** [comment]
@iamareebjamal Please Review now , Added the `server_default`

**Pull request merged**